### PR TITLE
Add information about support for single-node clusters

### PR DESCRIPTION
**Pull request by jillian-maroket**
Related Jira issue: SURE-7147

Scope: 
- "Installation" > "Hardware and Network Requirements": I reorganized existing cluster-related information and added a paragraph about single-node clusters.
- "Advanced" > "Single-Node Clusters": I added a file that contains information such as limitations and prerequisites. I plan to add content as I explore the link between Harvester and Longhorn (e.g., monitoring volumes).

### CODE REVIEW

File: docs/advanced/singlenodeclusters.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+sidebar_position: 7
+sidebar_label: Single-Node Clusters
+title: "Single-Node Clusters"
+keywords:
+- Single Node
+Description: Support for deployment of single-node clusters
+---
+
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/v1.2/advanced/singlenodeclusters">
```
Comment (bk201) [inline docs/advanced/singlenodeclusters.md:12]:
```suggestion
  <link rel="canonical" href="https://docs.harvesterhci.io/v1.2/advanced/singlenodeclusters"/>
```

File: docs/advanced/singlenodeclusters.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+</head>
+
+Beginning with v1.2.0, Harvester is supporting single-node clusters for implementations that require minimal initial deployment resources and/or that can tolerate lower resiliency. You can create single-node clusters using the standard installation methods ([ISO](https://docs.harvesterhci.io/v1.3/install/index), [USB](https://docs.harvesterhci.io/v1.3/install/usb-install), and [PXE boot](https://docs.harvesterhci.io/v1.3/install/pxe-boot-install)).
+
+Single-node clusters support most Harvester features, including the creation of RKE2 clusters and node upgrades (with some limitations). However, this deployment type has the following key disadvantages:
+
+- No high availability: Errors and updates that require rebooting of the node cause downtime to running VMs.
+- No multi-replica support: Only one replica is created for each volume in Longhorn.
+- No live migration and zero-downtime support during upgrades
```
Comment (asettle) [inline docs/advanced/singlenodeclusters.md:21]:
```suggestion
- No live migration and zero-downtime support during upgrades.

File: docs/advanced/singlenodeclusters.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+keywords:
+- Single Node
+Description: Support for deployment of single-node clusters
+---
+
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/v1.2/advanced/singlenodeclusters">
+</head>
+
+Beginning with v1.2.0, Harvester is supporting single-node clusters for implementations that require minimal initial deployment resources and/or that can tolerate lower resiliency. You can create single-node clusters using the standard installation methods ([ISO](https://docs.harvesterhci.io/v1.3/install/index), [USB](https://docs.harvesterhci.io/v1.3/install/usb-install), and [PXE boot](https://docs.harvesterhci.io/v1.3/install/pxe-boot-install)).
```
Comment (asettle) [inline docs/advanced/singlenodeclusters.md:15]:
```suggestion
As of Harvester release v1.2.0, single-node clusters are supported for implementations that require minimal initial deployment resources or that can tolerate lower resiliency. You can create single-node clusters using the standard installation methods ([ISO](https://docs.harvesterhci.io/v1.3/install/index), [USB](https://docs.harvesterhci.io/v1.3/install/usb-install), and [PXE boot](https://docs.harvesterhci.io/v1.3/install/pxe-boot-install)).

File: docs/advanced/singlenodeclusters.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+Single-node clusters support most Harvester features, including the creation of RKE2 clusters and node upgrades (with some limitations). However, this deployment type has the following key disadvantages:
+
+- No high availability: Errors and updates that require rebooting of the node cause downtime to running VMs.
+- No multi-replica support: Only one replica is created for each volume in Longhorn.
+- No live migration and zero-downtime support during upgrades
+
+## Prerequisites/Requirements
```
Comment (asettle) [inline docs/advanced/singlenodeclusters.md:23]: Do we usually use the slash here? I'd suggest just using "Prerequisites"

File: docs/advanced/singlenodeclusters.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+Before you begin deploying your single-node cluster, ensure that the following requirements are fulfilled/addressed.
```
Comment (asettle) [inline docs/advanced/singlenodeclusters.md:25]:
```suggestion
Before you begin deploying your single-node cluster, ensure that the following requirements are addressed.

File: docs/advanced/singlenodeclusters.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+- Hardware: [Use server-class hardware](https://docs.harvesterhci.io/v1.3/install/requirements#hardware-requirements) with sufficient resources to run Harvester and a production workload. Laptops and nested virtualization are not supported.
+- Network: [Configure ports](https://docs.harvesterhci.io/v1.3/install/requirements#port-requirements-for-harvester-nodes) based on the type of traffic to be transmitted among VMs.
+- StorageClass: [Create a new default StorageClass](https://docs.harvesterhci.io/v1.3/advanced/storageclass#creating-a-storageclass) with the **Number of Replicas** parameter set to "1". This ensures that only one replica is created for each volume in Longhorn.
+
+    :::info important
+    The default StorageClass "harvester-longhorn" has a replica count value of "3" for high availability. If you use this StorageClass to create volumes for your single-node cluster, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as "Degraded" on the Longhorn UI.
```
Comment (asettle) [inline docs/advanced/singlenodeclusters.md:32]:
```suggestion
    The default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes for your single-node cluster, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as *Degraded* on the Longhorn UI.

File: versioned_docs/version-v1.2/advanced/singlenodeclusters.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+keywords:
+- Single Node
+Description: Support for deployment of single-node clusters
+---
+
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/v1.2/advanced/singlenodeclusters">
+</head>
+
+Beginning with v1.2.0, Harvester is supporting single-node clusters for implementations that require minimal initial deployment resources and/or that can tolerate lower resiliency. You can create single-node clusters using the standard installation methods ([ISO](https://docs.harvesterhci.io/v1.2/install/index), [USB](https://docs.harvesterhci.io/v1.2/install/usb-install), and [PXE boot](https://docs.harvesterhci.io/v1.2/install/pxe-boot-install)).
```
Comment (asettle) [inline versioned_docs/version-v1.2/advanced/singlenodeclusters.md:15]: I'd recommend the same edits here as in the above file

File: versioned_docs/version-v1.2/advanced/singlenodeclusters.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+sidebar_position: 7
+sidebar_label: Single-Node Clusters
+title: "Single-Node Clusters"
+keywords:
+- Single Node
+Description: Support for deployment of single-node clusters
+---
+
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/v1.2/advanced/singlenodeclusters">
```
Comment (LucasSaintarbor) [inline versioned_docs/version-v1.2/advanced/singlenodeclusters.md:12]:
```suggestion
  <link rel="canonical" href="https://docs.harvesterhci.io/v1.2/advanced/singlenodeclusters"/>
```

File: docs/advanced/singlenodeclusters.md (hunk 4 above)
Comment (jillian-maroket) [reply docs/advanced/singlenodeclusters.md:23]: I'll use "Prerequisites" as suggested.

File: docs/advanced/singlenodeclusters.md (hunk 6 above)
Comment (jillian-maroket) [reply docs/advanced/singlenodeclusters.md]: I applied the suggested markup changes.

**Review by asettle — APPROVED** [review]

**Review by innobead — APPROVED** [review]

File: docs/advanced/singlenodeclusters.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+Before you begin deploying your single-node cluster, ensure that the following requirements are addressed.
+
+- Hardware: [Use server-class hardware](https://docs.harvesterhci.io/v1.3/install/requirements#hardware-requirements) with sufficient resources to run Harvester and a production workload. Laptops and nested virtualization are not supported.
+- Network: [Configure ports](https://docs.harvesterhci.io/v1.3/install/requirements#port-requirements-for-harvester-nodes) based on the type of traffic to be transmitted among VMs.
+- StorageClass: [Create a new default StorageClass](https://docs.harvesterhci.io/v1.3/advanced/storageclass#creating-a-storageclass) with the **Number of Replicas** parameter set to "1". This ensures that only one replica is created for each volume in Longhorn.
+
+    :::info important
+    The default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes for your single-node cluster, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as "Degraded" on the Longhorn UI.
+    :::
```
Comment (LucasSaintarbor) [inline docs/advanced/singlenodeclusters.md:33]:
```suggestion
  :::info important
  The default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes for your single-node cluster, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as "Degraded" on the Longhorn UI.
  :::
```

File: versioned_docs/version-v1.2/advanced/singlenodeclusters.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+Before you begin deploying your single-node cluster, ensure that the following requirements are addressed.
+
+- Hardware: [Use server-class hardware](https://docs.harvesterhci.io/v1.2/install/requirements#hardware-requirements) with sufficient resources to run Harvester and a production workload. Laptops and nested virtualization are not supported.
+- Network: [Configure ports](https://docs.harvesterhci.io/v1.2/install/requirements#port-requirements-for-harvester-nodes) based on the type of traffic to be transmitted among VMs.
+- StorageClass: [Create a new default StorageClass](https://docs.harvesterhci.io/v1.2/advanced/storageclass#creating-a-storageclass) with the **Number of Replicas** parameter set to "1". This ensures that only one replica is created for each volume in Longhorn.
+
+    :::info important
+    The default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes for your single-node cluster, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as "Degraded" on the Longhorn UI.
+    :::
```
Comment (LucasSaintarbor) [inline versioned_docs/version-v1.2/advanced/singlenodeclusters.md:33]:
```suggestion
  :::info important
  The default StorageClass `harvester-longhorn` has a replica count value of `3` for high availability. If you use this StorageClass to create volumes for your single-node cluster, Longhorn is unable to create the configured number of replicas. This results in volumes being marked as "Degraded" on the Longhorn UI.
  :::
```

**Review by LucasSaintarbor — CHANGES_REQUESTED** [review]
Updated the note formatting. LGTM otherwise! 

Screenshot 2023-12-12 at 8 40 31 AM

Screenshot 2023-12-12 at 8 39 34 AM

**Review by LucasSaintarbor — APPROVED** [review]
LGTM! 👍

**Pull request merged**